app/views/MainTemplate.scala

```diff
@@ -0,0 +1,136 @@
+import scala.concurrent.Await
+import scala.concurrent.duration.{Duration, SECONDS}
+
+@ImplementedBy(classOf[MainTemplateImpl])
+trait MainTemplate {
+
+  def apply(
+    pageTitle: String,
+    backLinkHref: Option[String] = None,
+    disableSessionExpired: Boolean = false,
+    additionalScripts: Option[Html] = None,
+    backLinkAttrs: Map[String, String] = Map.empty,
+    actingAttorney: Option[ActingAsAttorneyFor] = None,
+    beforeContentHtml: Option[Html] = None,
+    pageHeading: String = null,
+    showBackLink: Boolean = true,
+    headerSectionNeeded: Boolean = false
+  )(contentBlock: Html)(implicit
+    request: Request[_],
+    messages: Messages
+  ): HtmlFormat.Appendable
+}
+
+class MainTemplateImpl @Inject() (
+  appConfig: ApplicationConfig,
+  featureFlagService: FeatureFlagService,
+  wrapperService: WrapperService,
+  oldLayout: nonScaWrapperMain,
+  headBlock: HeadBlock,
+  sidebar: sidebar,
+  scripts: AdditionalJavascript
+) extends MainTemplate
+    with Logging {
+  override def apply(
+    pageTitle: String,
+    backLinkHref: Option[String] = None,
+    disableSessionExpired: Boolean = false,
+    additionalScripts: Option[Html] = None,
+    backLinkAttrs: Map[String, String] = Map.empty,
+    actingAttorney: Option[ActingAsAttorneyFor] = None,
+    beforeContentHtml: Option[Html] = None,
+    pageHeading: String = null,
+    showBackLink: Boolean = true,
+    headerSectionNeeded: Boolean = false
+  )(contentBlock: Html)(implicit request: Request[_], messages: Messages): HtmlFormat.Appendable = {
+
+    val scaWrapperToggle =
+      Await.result(featureFlagService.get(SCAWrapperToggle), Duration(appConfig.SCAWrapperFutureTimeout, SECONDS))
+    val fullPageTitle    = s"$pageTitle - ${Messages("generic.ats.browser.title")}"
+
+    if (scaWrapperToggle.isEnabled) {
+      logger.debug(s"SCA Wrapper layout used for request `${request.uri}``")
+
+      val showAccountMenu = actingAttorney.isEmpty && !disableSessionExpired
+
+      wrapperService.layout(
+        content = contentBlock,
+        pageTitle = Some(fullPageTitle),
+        serviceNameKey = Some(messages("generic.ats")),
+        serviceNameUrl = Some(appConfig.homePageUrl),
+        sidebarContent = Some(sidebar(beforeContentHtml)),
+        signoutUrl = controllers.routes.AccountController.signOut.url,
+        timeOutUrl = Some(controllers.routes.AccountController.sessionExpired.url),
+        keepAliveUrl = controllers.routes.AccountController.keepAlive.url,
+        styleSheets = Seq(headBlock()),
+        bannerConfig = BannerConfig(
+          showAlphaBanner = false,
+          showBetaBanner = true,
+          showHelpImproveBanner = true
+        ),
+        fullWidth = false,
+        hideMenuBar = !showAccountMenu,
+        disableSessionExpired = disableSessionExpired,
+        backLinkUrl = backLinkHref,
+        showBackLinkJS = if (backLinkHref.isDefined) {
+          true
+        } else {
+          false
+        },
```

Comment:
Small change but you can get rid of the if/else and just do showBackLinkJS = backLinkHref.isDefined

Reply:
Ah wait, should it be reversed actually, showBackLinkJs = !backLinkHref.isDefined? Otherwise had coded back links won't work

Reply:
backLinkHref.isDefined is the correct one. There aren't any hard coded back link as such that goes to particular page, all the back links points to "" or #. I will get rid of if else